src/script/lua_api/l_env.cpp

```diff
@@ -340,6 +340,31 @@ int ModApiEnv::l_swap_node(lua_State *L)
 	return 1;
 }
 
+// bulk_swap_node([pos1, pos2, ...], node)
+// pos = {x=num, y=num, z=num}
+int ModApiEnv::l_bulk_swap_node(lua_State *L)
+{
+	GET_ENV_PTR;
+
+	luaL_checktype(L, 1, LUA_TTABLE);
+
+	s32 len = lua_objlen(L, 1);
+
+	MapNode n = readnode(L, 2);
+
+	// Do it
+	bool succeeded = true;
+	for (s32 i = 1; i <= len; i++) {
+		lua_rawgeti(L, 1, i);
+		if (!env->swapNode(read_v3s16(L, -1), n))
+			succeeded = false;
```

Comment:
i'm not sure about the return, we will say all fail for some only. One short term option can be to return error count instead , or the list of failed nodes coordinates

Reply:
it's maybe because it almost never fail and people never care about handling the fact it failed ?